### PR TITLE
Added tests for restoreState and updateBreakpoints

### PR DESCRIPTION
**Pull request by JohanMabille**

### CODE REVIEW

File: tests/src/service.spec.ts [hunk 1]
```diff
@@ -91,4 +93,58 @@ describe('DebugService', () => {
...
+            path: source_id
+          }
+        };
+      });
+      await service.updateBreakpoints(code, breakpoints);
+    });
+
+    describe('#updateBreakpoints', () => {
+      it('should update the breakpoints', () => {
+        const bp_list = model.breakpointsModel.getBreakpoints(source_id);
```
Comment (jtpio) [inline tests/src/service.spec.ts]: camelCase for the naming of variables? :)

Comment (JohanMabille) [reply tests/src/service.spec.ts]: I did too much C++ during the weekend

Comment (JohanMabille) [reply tests/src/service.spec.ts]: I did too much C++ during the weekend

**Comment by jtpio** [comment]
@JohanMabille could this change also include https://github.com/jupyterlab/debugger/issues/166? or should this be done separately?

**Comment by JohanMabille** [comment]
Ah good catch. I'll add it.

File: src/tokens.ts [hunk 2]
```diff
@@ -85,7 +85,7 @@ export interface IDebugger extends IDisposable {
 
   /**
    * Restart the debugger.
-   * Precondition: isStarted()
+   * Precondition: !isStarted()
```
Comment (jtpio) [inline src/tokens.ts]: Is that true?

Comment (JohanMabille) [reply src/tokens.ts]: No, that's a wrong update.

File: src/service.ts [hunk 3]
```diff
@@ -194,12 +194,10 @@ export class DebugService implements IDebugger {
     // No need to dump the cells again, we can simply
     // resend the breakpoints to the kernel and update
     // the model.
-    breakpoints.forEach(async (bp, path, _) => {
-      const sourceBreakpoints = Private.toSourceBreakpoints(bp);
-      await this.setBreakpoints(sourceBreakpoints, path);
-    });
-
-    this.model.breakpointsModel.restoreBreakpoints(breakpoints);
+    for (const bp of breakpoints) {
```
Comment (jtpio) [inline src/service.ts]: How about something like this?

```suggestion
    for (const [source, bps] of breakpoints) {
```

Comment (JohanMabille) [reply src/service.ts]: definitely better!

**Comment by jtpio** [comment]
This fixes #166.

**Pull request merged**